File: app/src/main/java/com/example/breezil/favnews/di/module/SettingsActivityModule.java
package com.example.breezil.favnews.di.module;

import com.example.breezil.favnews.ui.SettingsActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;


@Module
public abstract class SettingsActivityModule {
    @ContributesAndroidInjector(modules = SettingsFragmentsModule.class)
    abstract SettingsActivity contributeSettingsActivity();
}

File: app/src/main/java/com/example/breezil/fave/di/module/SettingsActivityModule.kt
package com.example.breezil.fave.di.module

import com.example.breezil.fave.ui.preference.SettingsActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
abstract class SettingsActivityModule {
    @ContributesAndroidInjector(modules = [SettingsFragmentsModule::class])
    internal abstract fun contributeSettingsActivity(): SettingsActivity
}